lib/vanagon/engine/amazon.rb

```diff
@@ -0,0 +1,69 @@
+require 'aws-sdk'
+require 'erb'
+require 'base64'
+require 'vanagon/engine/base'
+
+class Vanagon
+  class Engine
+    class Amazon < Base
+      attr_accessor :ami, :key_name, :userdata, :key, :key_name, :shutdown_behavior
+      attr_accessor :subnet_id, :instance_type, :vpc_id
+
+      def initialize(platform, target = nil)
+        super
+        @name = 'amazon'
+        @platform = platform
+        @ami = @platform.aws_ami
+        @target_user = @platform.target_user
+        @subnet_id = @platform.aws_subnet_id
+        @userdata = @platform.aws_user_data
+        @region = @platform.aws_region.nil? ? 'us-east-1' : @platform.region
+        @key_name = @platform.aws_key_name
+        @key = @platform.aws_key
+        @vpc_id = @platform.aws_vpc_id
+        @instance_type = @platform.aws_instance_type.nil? ? "t1.micro" : @platform.aws_instance_type
+        @required_attributes = ["ssh_port", "aws_ami", "aws_key_name"]
+        @shutdown_behavior = @platform.aws_shutdown_behavior
+
+        @ec2 = ::Aws::EC2::Client.new region: @region
```

Comment:
This is hard to codify without an explicit style guide (which I think we're actually working on), but we try to use parenthesis when constructing assignments like this:

```ruby
@ec2 = ::Aws::EC2::Client.new(region: @region)
```